### PR TITLE
Enable container and Kubernetes awareness for improved telemetry.

### PR DESCRIPTION
**Pull request by vbabanin**
Include container and Kubernetes awareness, as specified in the [MongoDB Handshake Specification](https://github.com/abr-egn/specifications/blob/c4831b426c7eab8749723d7f3cd6bb5c237bcbbe/source/mongodb-handshake/handshake.rst#container)

JAVA-5072

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -257,6 +257,7 @@ configure(javaCodeCheckedProjects) {
         testImplementation 'org.spockframework:spock-core'
         testImplementation 'org.spockframework:spock-junit4'
         testImplementation("org.mockito:mockito-core:3.8.0")
+        testImplementation("org.mockito:mockito-inline:3.8.0")
```
Comment (vbabanin) [inline build.gradle:260]: This useful addon to Mockito allows for the mocking of static methods.

File: driver-core/src/main/com/mongodb/internal/connection/ClientMetadataHelper.java [hunk 2]
```diff
@@ -168,8 +180,7 @@ static boolean clientMetadataDocumentTooLarge(final BsonDocument document) {
         new BsonDocumentCodec().encode(new BsonBinaryWriter(buffer), document, EncoderContext.builder().build());
         return buffer.getPosition() > MAXIMUM_CLIENT_METADATA_ENCODED_SIZE;
     }
-
-    private enum Environment {
+    private enum FaasEnvironment {
```
Comment (vbabanin) [inline driver-core/src/main/com/mongodb/internal/connection/ClientMetadataHelper.java:183]: With the introduction of new parameters in the 'env' section related to containers/orchestration, I've renamed this enum in accordance with the [specification](https://github.com/abr-egn/specifications/blob/c4831b426c7eab8749723d7f3cd6bb5c237bcbbe/source/mongodb-handshake/handshake.rst#faas) as contains metadata for FaaS.

Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/connection/ClientMetadataHelper.java:183]: Ack

File: build.gradle (hunk 1 above)
Comment (rozza) [reply build.gradle:260]: Nice

**Review by rozza — APPROVED** [review]
LGTM

**Pull request merged**